improver/ensemble_copula_coupling/utilities.py

```diff
@@ -242,6 +237,8 @@ def insert_lower_and_upper_endpoint_to_1d_array(array_1d, low_endpoint, high_end
         numpy.ndarray:
             1d array of values padded with the low_endpoint and high_endpoint.
     """
+    if array_1d.ndim != 1:
+        raise ValueError("Expected 1D input, got {}".format(array_1d.ndim))
```

Comment:
Possibly this could be:
```suggestion
        raise ValueError("Expected 1D input, got {}D input".format(array_1d.ndim))
```